lib/cmd_grepip.go

```diff
@@ -121,14 +121,12 @@ func CmdGrepIP(
 
 	// prepare regexp
 	var rexp *regexp.Regexp
-	rexp4 := `((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)`
-	rexp6 := `(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))`
 	if ipv == 4 {
-		rexp = regexp.MustCompilePOSIX(rexp4)
+		rexp = regexp.MustCompilePOSIX(ipV4RgxPattern)
 	} else if ipv == 6 {
-		rexp = regexp.MustCompilePOSIX(rexp6)
+		rexp = regexp.MustCompilePOSIX(ipV4RgxPattern)
 	} else {
-		rexp = regexp.MustCompilePOSIX(rexp4 + "|" + rexp6)
+		rexp = regexp.MustCompilePOSIX(ipV4RgxPattern + "|" + ipV6RgxPattern)
```

Comment:
No need to even compile it here if we compiled it before at the global level - we can just assign `rexp` to be the 'correct' global regexp variable